src/state/network/updater.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect } from 'react'
+import { useLocation } from 'react-router'
+
+import { Network } from 'types'
+import useGlobalState from 'hooks/useGlobalState'
+
+import { setNetwork } from './actions'
+import { useNetworkId } from './hooks'
+
+function getNetworkId(network: string | undefined): Network {
+  switch (network) {
+    case 'rinkeby':
+      return Network.Rinkeby
+    case 'xdai':
+      return Network.xDAI
+    default:
+      return Network.Mainnet
+  }
+}
+
+export const NetworkUpdater: React.FC = () => {
```

Comment:
Doesn't have children nor takes in any params. Use a simple function instead:

```suggestion
export function NetworkUpdater(): JSX.Element {
```

Reply:
BTW, does it have to be a component? Can't we use it as a hook instead?

In my mind components are things that display something. Feels weird to have a component that'll always return `null`.

Reply:
Fair enough! I followed uniswap approach, but I think u are right, it feels more natural to use a hook. Addressed